### ISSUE TITLE
Subscribing for location changes while app is suspended in background.

### ISSUE THREAD

**Issue by sm1ddy** [NativeScript/nativescript-geolocation#44]
Is it possible for a Nativescript app to use geolocation while an app is in the background or suspended. For example, Apple provide a Significant-Change Location Service that from my understanding "wakes" the app up for an event?

https://developer.apple.com/library/content/documentation/UserExperience/Conceptual/LocationAwarenessPG/CoreLocation/CoreLocation.html

Any examples or guidance out there? I've had trouble finding any :)
Basically, I'm trying to make an app that will prompt the user when they encounter a specific location (even if the app isn't currently open).

**Comment by sm1ddy** [comment]
For anyone looking at this issue, I incorporated the following into my nativescript-geolocation (which seems to work for the significant location api). It's pretty much a modified copy of watchLocation:

> function checkLocation(successCallback, errorCallback, options) {
>     var locListener = LocationListenerImpl.initWithLocationError(successCallback, errorCallback);
>     try {
>         var iosLocManager = LocationMonitor.createiOSLocationManager(locListener, options);
>         iosLocManager.startMonitoringSignificantLocationChanges();
>         return locListener.id;
>     }
>     catch (e) {
>         LocationMonitor.stopLocationMonitoring(locListener.id);
>         errorCallback(e);
>         return null;
>     }
> }
> exports.checkLocation = checkLocation;

**Comment by TLmaK0** [comment]
We should add:

`iosLocManager.allowsBackgroundLocationUpdates = true;`

https://developer.apple.com/reference/corelocation/cllocationmanager/1620568-allowsbackgroundlocationupdates
https://github.com/apache/cordova-plugin-geolocation/pull/51

**Comment by MrVonkey** [comment]
@TLmaK0  There is a pull request for this issue: #49

**Comment by DimitarTachev** [comment]
The @MrVonkey's changes were merged and the option is available in [v4.1.0](https://github.com/NativeScript/nativescript-geolocation/releases/tag/v4.1.0).
